base_layer/wallet_ffi/src/lib.rs

```diff
@@ -244,6 +244,40 @@ pub struct TariWallet {
     shutdown: Shutdown,
 }
 
+#[derive(Debug)]
+#[repr(C)]
```

Comment:
Look at the c code in the .h file we need accessor functions for the node id handling the `Option<TariNodeId>`
And the u128 is not support in C, its only fully supported in GCC c23 which had its draft published in Jan 2023.